exec/java-exec/src/main/java/org/apache/drill/exec/client/DrillClient.java

```diff
@@ -245,14 +291,15 @@ public synchronized void connect(String connect, Properties props) throws RpcExc
           throw new RpcException("Failure setting up ZK for client.", e);
         }
       }
-
-      final ArrayList<DrillbitEndpoint> endpoints = new ArrayList<>(clusterCoordinator.getAvailableEndpoints());
-      checkState(!endpoints.isEmpty(), "No DrillbitEndpoint can be found");
-      // shuffle the collection then get the first endpoint
-      Collections.shuffle(endpoints);
-      endpoint = endpoints.iterator().next();
+      endpoints.addAll(clusterCoordinator.getAvailableEndpoints());
     }
 
+    // Make sure we have at least one endpoint in the list
+    checkState(!endpoints.isEmpty(), "No DrillbitEndpoint can be found");
```

Comment:
If this error is shown to the user, perhaps help them out a bit:

No Drillbit endpoints found in either the connect string (drillbit=host[:port]) or in the drill-override.conf file.